internal/processor/data.go

```diff
@@ -60,3 +69,44 @@ func processDataSet(dataSet *map[string]interface{}, samplesToMerge *load.Sample
 		CreateMetricSets(mergedData, cfg, i, true, samplesToMerge, originalAPINo)
 	}
 }
+
+func runJq(dataSets interface{}, api load.API) []interface{} {
+	// due to the current design, dataSets is always a slice
+	// so an api that returns a map will still be a slice returned for processing
+	// for better usability if there is no array prefix access the first item in the slice automatically for the user
+	if !strings.HasPrefix(api.Jq, ".[") {
+		api.Jq = fmt.Sprintf(".[0]%v", api.Jq)
+	}
+
+	query, err := gojq.Parse(api.Jq)
+	if err != nil {
+		load.Logrus.WithFields(logrus.Fields{
+			"api": api.Name,
+		}).WithError(err).Error("jq: failed to parse")
+		return []interface{}{}
+	}
+
+	iter := query.Run(dataSets)
+	for {
+		v, ok := iter.Next()
+		if !ok {
+			break
+		}
+
+		switch value := v.(type) {
+		case []interface{}:
+			return value
+		case map[string]interface{}:
+			return []interface{}{value}
+		case error:
+			load.Logrus.WithFields(logrus.Fields{
+				"api": api.Name,
+				"jq":  api.Jq,
+			}).WithError(err).Debug("jq: failed to process")
+			return []interface{}{}
```

Comment:
@Kav91 
- I always have this dilemma: 

1. do we return the original dataSets when error
2. do we return empty data

Reply:
@haihongren was experiencing the same dilemma.
I felt that if someone was using the jq functionality, and it didn't do its job we shouldn't return the dataset. 
So at least they can go look at the error and realize it was because of that. Otherwise if we return and they do subsequent transformations the output would be wrong anyway.

Reply:
@kav91 Agreed.  Nice job adding built-in `jq` support in Flex!